static/js/main.js

```diff
@@ -79,9 +99,27 @@ function requestNotificationPermission() {
     .then(function (permissionResult) {
         if (permissionResult === 'granted') {
             console.log("Notification Permission granted!")
-            console.log("TODO: subscribe user to push notification!")
+            return executeUserSubscriptionFlow()
         } else {
             throw new Error("Notification Permission NOT granted.")
         }
     }).catch((error) => logError(error))
 }
+
+function executeUserSubscriptionFlow() {
+    registerServiceWorker()
+    .then(function (registration) {
```

Comment:
same comment for arrow function as previous PR, it will look a little uniform if you want to make function like this.
but looks good,